nessie/src/test/java/org/apache/iceberg/nessie/TestBranchVisibility.java

```diff
@@ -173,6 +174,33 @@ public void testSchemaSnapshot() throws Exception {
     Assertions.assertThat(metadataOn2).isNotEqualTo(metadataOnTest).isNotEqualTo(metadataOnTest2);
   }
 
+  @Test
+  public void testMetadataLocationWithRefresh() throws Exception {
+    String branch1 = "test";
+    String branch2 = "branch-2";
+
+    // commit on tableIdentifier1 on branch1
+    NessieCatalog catalog = initCatalog(branch1);
+    String metadataCommit1 = addRow(catalog, tableIdentifier1, "initial-data",
+        ImmutableMap.of("id0", 4L));
+    catalog.refresh();
```

Comment:
those `catalog.refresh()` calls can be removed

Reply:
it is just a double check that even refresh is not gonna fix it.

Reply:
refresh() shouldn't be used outside of the internals of the catalog imo (I have a PR that cleans that up). Also it's not apparent that this is the reason why refresh() is being used here, so I woudl just remove it

Reply:
ok. Removed.